File: js/task3a/S3_OBJ_06.js
function groupBy(items, key) {
  const result = {};

  for (const item of items) {
    const value = item[key];
    if (!result[value]) {
      result[value] = [];
    }
    result[value].push(item);
  }
  return result;
}

const people = [
  { name: "Ana", city: "Lima" },
  { name: "Luis", city: "Cusco" },
  { name: "Pedro", city: "Lima" }
];

console.log(groupBy(people, "city"));

File: python/task3a/S3_OBJ_06.py
def group_by(items, key):
    result = {}
    for item in items:
        k = item.get(key)
        result.setdefault(k, []).append(item)
    return result

users = [{"name":"Ana","role":"admin"},{"name":"Bob","role":"user"}]
print(group_by(users, "role"))
